include/roboteam_ai/world/BallPossession.h

```diff
@@ -11,33 +7,89 @@
 namespace rtt {
 namespace ai {
 
+/**
+ * Computes and stores the ball possession state which indicates which team controls the ball (can also be both/no team).
+ * @author Created by: Rolf van der Hulst <br>
+ *         Documented by: Haico Dorenbos
+ * @since 2019-04-15
+ */
 class BallPossession {
     FRIEND_TEST(BallPossessionTest, team_far_or_close_to_ball);
     FRIEND_TEST(BallPossessionTest, it_properly_computes);
+
+private:
+    const double MIDDLE_LINE_X = 0.0; // The x coordinate that corresponds to the middle of the field.
+    /* If a teams distance to the ball is larger or equal than STANDARD_FAR_THRESHOLD at a moment then that team is
+     * considered to be far from the ball at that moment. */
+    const double STANDARD_FAR_THRESHOLD = 0.4;
+    /* Use DEFENSIVE_FAR_THRESHOLD instead of STANDARD_FAR_THRESHOLD for the opponents team if the ball is at our side
+     * of the field. In this case we should play more defensively and therefore the opponent is only considered to be
+     * far from the ball if the opponent has a larger distance to the ball than in the general case. */
+    const double DEFENSIVE_FAR_THRESHOLD = 0.9;
+    /* If the ball has been uninterruptedly close to a team for this number of seconds then this team is considered to
+     * be long enough close to the ball. */
+    const double CLOSE_TIME_TRESHOLD = 0.05;
+    /* If the ball has been uninterruptedly far from a team for this number of seconds then this team is considered
+     * to be long enough far from the ball. */
+    const double FAR_TIME_TRESHOLD = 1.5;
+    /* If the ball x coordinate is larger than relatively OUR_POSSESSION_RELATIVE_WIDTH_THRESHOLD of the width of the
+     * field (with respect to the leftmost x-coordinate of the field) then the ball is considered to be always in our
+     * possession. */
+    const double OUR_POSSESSION_RELATIVE_WIDTH_THRESHOLD = 0.75; // 6/8 of the field
+    /* If the ball x coordinate is smaller than relatively THEIR_POSSESSION_RELATIVE_WIDTH_THRESHOLD of the width of
+     * the field (with respect to the leftmost x-coordinate of the field) then the ball is always considered to be in
+     * their possession. */
+    const double THEIR_POSSESSION_RELATIVE_WIDTH_THRESHOLD = 0.375; // 3/8 of the field
+
 public:
     enum Possession {
-        LOOSEBALL,
-        OURBALL,
-        THEIRBALL,
-        CONTENDEDBALL
+        LOOSEBALL,      // If neither us nor the opponent possesses the ball.
+        OURBALL,        // If we possesses the ball.
+        THEIRBALL,      // If the opponent possesses the ball.
+        CONTENDEDBALL   // If we and the opponent both possesses the ball.
```

Comment:
I do not agree: contended means challenged according to the Cambridge dictionary. And because the meaning is well documented this should not be a problem.